samples/readSnippets.js

```diff
@@ -0,0 +1,218 @@
+      .on('data', row => printRow(row.id, row.data))
+      .on('end', () => {
+        // All rows retrieved.
+      });
+  }
+
+  // [END bigtable_reads_row_range]
+
+  // [START bigtable_reads_row_ranges]
+  async function readRowRanges() {
+    await table
+      .createReadStream({
+        ranges: [
+          {
+            start: 'phone#4c410523#20190501',
+            end: 'phone#4c410523#20190601',
+          },
+          {
+            start: 'phone#5c10102#20190501',
+            end: 'phone#5c10102#20190601',
+          },
+        ],
+      })
+      .on('error', err => {
+        // Handle the error.
+        console.log(err);
+      })
+      .on('data', row => printRow(row.id, row.data))
+      .on('end', () => {
+        // All rows retrieved.
+      });
+  }
+
+  // [END bigtable_reads_row_ranges]
+
+  // [START bigtable_reads_prefix]
+  async function readPrefix() {
+    const prefix = 'phone#';
+
+    await table
+      .createReadStream({
+        prefix,
+      })
+      .on('error', err => {
+        // Handle the error.
+        console.log(err);
+      })
+      .on('data', row => printRow(row.id, row.data))
+      .on('end', () => {
+        // All rows retrieved.
+      });
+  }
+
+  // [END bigtable_reads_prefix]
+
+  // [START bigtable_reads_filter]
+  async function readFilter() {
+    const filter = {
+      value: /PQ2A.*$/,
+    };
+
+    await table
+      .createReadStream({
+        filter,
+      })
+      .on('error', err => {
+        // Handle the error.
+        console.log(err);
+      })
+      .on('data', row => printRow(row.id, row.data))
+      .on('end', () => {
+        // All rows retrieved.
+      });
+  }
+
+  // [END bigtable_reads_filter]
+
+  // [START bigtable_reads_row]
+  // [START bigtable_reads_row_partial]
+  // [START bigtable_reads_rows]
+  // [START bigtable_reads_row_range]
+  // [START bigtable_reads_row_ranges]
+  // [START bigtable_reads_prefix]
+  // [START bigtable_reads_filter]
+  function printRow(rowkey, rowData) {
+    console.log(`Reading data for ${rowkey}:`);
+
+    for (const columnFamily of Object.keys(rowData)) {
+      const columnFamilyData = rowData[columnFamily];
+      console.log(`Column Family ${columnFamily}`);
+
+      for (const columnQualifier of Object.keys(columnFamilyData)) {
+        const col = columnFamilyData[columnQualifier];
+
+        for (let i = 0; i < col.length; i++) {
+          const cell = col[i];
+          const labels = cell.labels.length
+            ? ` [${cell.labels.join(',')}]`
+            : '';
+          console.log(
+            `\t${columnQualifier}: ${cell.value} @${cell.timestamp}${labels}`
+          );
+        }
+      }
+    }
+    console.log();
+  }
+
+  // [END bigtable_reads_row]
+  // [END bigtable_reads_row_partial]
+  // [END bigtable_reads_rows]
+  // [END bigtable_reads_row_range]
+  // [END bigtable_reads_row_ranges]
+  // [END bigtable_reads_prefix]
+  // [END bigtable_reads_filter]
+
+  eval(`${read}()`);
```

Comment:
I would avoid using `eval()` in general it's frowned upon by the community because it can lead to vulnerabilities in code, as an example, this would allow me to run:

```
node readSnippets.js instance-id table-id "require('rimraf').sync('/nuke/my/computer')"
```

This is also resulting in `linting` issues since it sees the method names defined but never called. As an alternative implementation, I would just do a `switch` statement ... this will correct the linting error, and make security people's eyes bulge less :laughing: 

```js
switch (read) {
  caset "printRow":
    printRow();
    break;
}
```

I know, way less cool.

Reply:
Alright! I actually had it that way originally, but was requested to do it as functions in Java, so I was just trying to keep things consistent. And yeah, I know about the one sample per file, I wanted to reuse the print functions and other logic since it is so repetitive here.